### ISSUE TITLE
No intellisense for node JS modules not using relative require path

### ISSUE THREAD

**Issue by dakaraphi** [Microsoft/vscode#37056]
- VSCode Version: 1.17.2
- OS Version:

Steps to Reproduce:

Sample directory structure:
![image](https://user-images.githubusercontent.com/10656375/32121771-f731ec42-bb2b-11e7-9fa6-8eaf39fff035.png)

The following require statements labelled with comments do not have intellisense.
![image](https://user-images.githubusercontent.com/10656375/32121743-deda9860-bb2b-11e7-94f7-c56d2a7e7370.png)

These are all plain JS node modules.  It seems that require only works when a relative path is specified, but is broken for non relative import.
I added `ts-check` to verify the modules are found, and that seems to be true.

Reproduces without extensions: Yes

**Comment by mjbvz** [comment]
If you create a `jsconfig.json` at the root of your project with the contents:

```
{
    "compilerOptions": {
        "target": "ES6",
        "module": "commonjs"
    },
    "exclude": [
        "node_modules",
        "**/node_modules/*"
    ]
}
```

Do you see the expected behavior? The important configuration option is `"module": "commonjs"`

**Comment by dakaraphi** [comment]
@mjbvz Thanks! that resolves my issue.

**Comment by dakaraphi** [comment]
@mjbvz unfortunately this only fixed it for this example.
In my real project, it made it work only intermittently.

I'm getting really strange results.  Like I can create a new JS file and a require statement can not find any non relative module.  If I then restart VS code it will work for that file.  I then have to do this for each new file I create.  Sometimes files that were working stop working.

Could this be related in any way to the scale of the project?  Just the portion I'm testing is over 2k files.  The full project is probably about 10k files.  It keeps telling me to enable the `excludes` in my jsconfig, but they are already configured.

**Comment by mjbvz** [comment]
Yes, if you are seeing the configure excludes message, intellisense will have been disable for some files in your project. It will also be disabled for all new files that are added. https://github.com/Microsoft/TypeScript/issues/19326 tracks better support for large project

Besides defining a smaller project, you can also try checking to make sure that all files in that TypeScript is loading as part of your project should really be part of your project:

1. Set `"typescript.tsserver.log": "verbose"` in vscode
1. Restart vscode and open your project
1. Run `TypeScript: Open TS Server log` in vscode
1. Open the `tsserver.log` file inside the directory that vscode opens

Look in the `tsserver.log` file for a long list of files towards the bottom. This is a list all the files that are currently be loading into your projects. Make sure these all make sense

**Comment by dakaraphi** [comment]
@mjbvz I have enabled the logging, but it is not clear to me from the logging what exactly is happening.  I have 2 main issues:

**No logging in my large project**
I can only get the logging to work if I move my JS out of the main project into a separate folder.
Not sure what the issue is, but the main project has 30k+ files.  It seems it is loading no files.  I get no warnings about configuring excludes.  I tried both adding and removing excludes.

**Logging, but not consistent with observations**
When I copy all JS out to a separate folder.  It partially works as I described above.  However, the logging doesn't seem to be consistent with the behavior I'm seeing.  
There are 2k+ JS files.  Whenever I open a file I see additions to the `tsserver.log` ;However, these additions are only some small portion of the project.  Possibly a hundred or so files appear to be open.
However, I don't get intellisense for all of the requires even though I see them listed as loaded in the `tsserver.log`.

If the log is not exceeding the 1k open file limit, shouldn't intellisense be working between all modules that the log show are loaded?
